encoder/vaapiencoder_vp9.cpp

```diff
@@ -191,23 +192,56 @@ bool VaapiEncoderVP9::fill(VAEncPictureParameterBufferVP9* picParam,
             picParam->reference_frames[i] = VA_INVALID_SURFACE;
     }
     else {
-        picParam->refresh_frame_flags = 0x01; // refresh last frame
+
         picParam->pic_flags.bits.frame_type = kInterFrame;
 
         ReferenceQueue::const_iterator it = m_reference.begin();
 
-        for (uint32_t i = 0; it != m_reference.end(); ++it, i++)
+        for (uint32_t i = 0; it != m_reference.end(); ++it, i++) {
             picParam->reference_frames[i] = (*it)->getID();
+#if __ENABLE_DEBUG__
+            DEBUG("reference frame[%d] 0x%x", i, (*it)->getID());
+#endif
+        }
 
+        picParam->pic_flags.bits.frame_context_idx = 0;
         // last/golden/alt is used as reference frame. L0 forward
         picParam->ref_flags.bits.ref_frame_ctrl_l0 = 0x7;
 
         // golden and alt are last KeyFrame
         // last is last decoded frame
-        picParam->ref_flags.bits.ref_last_idx = 0;
-        picParam->ref_flags.bits.ref_gf_idx = 1;
-        picParam->ref_flags.bits.ref_arf_idx = 2;
-        picParam->pic_flags.bits.frame_context_idx = 0;
+
+        if (getReferenceMode()) {
```

Comment:
In my view, this solution is simpler than creating an index array as I am working directly on the reference_frames deque.  m_currentReferenceIndex plays the role of the the index array with only one variable.  I did consider creating the index array and came up with this instead.